src/compiler/diagnosticMessages.json

```diff
@@ -703,6 +703,27 @@
         "category": "Error",
         "code": 1223
     },
+    "Signature '{0}' must have a type predicate.": {
+        "category": "Error",
+        "code": 1224
+    },
+    "Cannot find parameter '{0}'.": {
+        "category": "Error",
+        "code": 1225
+    },
+    "Type predicate '{0}' is not assignable to '{1}'.": {
+        "category": "Error",
+        "code": 1226
+    },
+    "Parameter '{0}' is not in the same position as parameter '{1}'.": {
+        "category": "Error",
+        "code": 1227
+    },
+    "Cannot define type predicate '{0}' as a type to a type predicate.": {

```

Comment:
You need a more general error that will disallow type predicates in all positions, except return type. A nested type predicate is just a specific case of that.